src/lib.rs

```diff
@@ -126,23 +127,69 @@ impl Debug for GetResult {
 impl GetResult {
     /// Collects the data into a [`Bytes`]
     pub async fn bytes(self) -> Result<Bytes> {
-        collect_bytes(self.into_stream(), None).await
+        match self {
+            Self::File(mut file, path) => {
+                maybe_spawn_blocking(move || {
+                    let len = file
+                        .seek(SeekFrom::End(0))
+                        .map_err(|source| local::Error::Seek {
+                            source,
+                            path: path.clone(),
+                        })?;
+
+                    file.seek(SeekFrom::Start(0))
+                        .map_err(|source| local::Error::Seek {
+                            source,
+                            path: path.clone(),
+                        })?;
+
+                    let mut buffer = Vec::with_capacity(len as usize);
+                    file.read_to_end(&mut buffer)
+                        .map_err(|source| local::Error::UnableToReadBytes { source, path })?;
+
+                    Ok(buffer.into())
+                })
+                .await
+            }
+            Self::Stream(s) => collect_bytes(s, None).await,
+        }
     }
 
     /// Converts this into a byte stream
+    ///
+    /// If the result is [`Self::File`] will perform chunked reads of the file, otherwise
+    /// will return the [`Self::Stream`].
+    ///
+    /// # Blocking Behaviour
+    ///
+    /// If called from a tokio context, will use [`tokio::runtime::Handle::spawn_blocking`] to
+    /// spawn the blocking work to a background thread, otherwise will perform potentially
+    /// blocking IO on the current thread
     pub fn into_stream(self) -> BoxStream<'static, Result<Bytes>> {
         match self {
             Self::File(file, path) => {
-                tokio_util::codec::FramedRead::new(file, tokio_util::codec::BytesCodec::new())
-                    .map_ok(|b| b.freeze())
-                    .map_err(move |source| {
-                        local::Error::UnableToReadBytes {
-                            source,
-                            path: path.clone(),
+                const CHUNK_SIZE: usize = 8 * 1024;
+
+                futures::stream::try_unfold((file, path, false), |(mut file, path, finished)| {
```

Comment:
TIL: https://docs.rs/futures/latest/futures/prelude/stream/fn.try_unfold.html# 🤔

Reply:
is it worth a comment explaining what the tuple is (file, path and "is done")?

Reply:
I think it is fairly clear given the rest of the line